test/e2e/main.go

```diff
@@ -46,66 +102,147 @@ func run() (string, error) {
+		ObjectMeta: metav1.ObjectMeta{
+			Name: *name,
+			Labels: map[string]string{
+				"test.mlkube.io": "",
+			},
+		},
+		Spec: torchv1alpha1.PyTorchJobSpec{
+			ReplicaSpecs: []*torchv1alpha1.PyTorchReplicaSpec{
+				torchReplicaType(torchv1alpha1.MASTER).toSpec(1),
+				torchReplicaType(torchv1alpha1.WORKER).toSpec(3),
+			},
+		},
+	}
+
+	code, err := ioutil.ReadFile("test/e2e/mnist.py")
+	if err != nil {
+		fmt.Print(err)
+	}
+
+	confCode := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dist-train",
+			Namespace: *namespace,
+		},
+		Data: map[string]string{
+			"dist_train.py": string(code),
+		},
+	}
+	_, err = client.CoreV1().ConfigMaps(*namespace).Create(confCode)
 	if err != nil {
 		log.Errorf("Creating the configmap failed; %v", err)
 		return *name, err
 	}
-	cmd = exec.Command(
-		"kubectl", "create",
-		"-f",
-		"examples/mnist/pytorchjob.yaml",
-		"-n",
-		*namespace,
-	)
-	err = runCmd(cmd)
+	// Create PyTorchJob
+	_, err = torchJobClient.KubeflowV1alpha1().PyTorchJobs(*namespace).Create(original)
 	if err != nil {
 		log.Errorf("Creating the job failed; %v", err)
 		return *name, err
 	}
 
 	// TODO(jose5918) Wait for completed state
 	// Wait for operator to reach running state
-	var tfJob *torchv1alpha1.PyTorchJob
+	var torchJob *torchv1alpha1.PyTorchJob
 	for endTime := time.Now().Add(*timeout); time.Now().Before(endTime); {
-		tfJob, err = tfJobClient.KubeflowV1alpha1().PyTorchJobs(*namespace).Get(*name, metav1.GetOptions{})
+		torchJob, err = torchJobClient.KubeflowV1alpha1().PyTorchJobs(*namespace).Get(*name, metav1.GetOptions{})
 		if err != nil {
 			log.Warningf("There was a problem getting PyTorchJob: %v; error %v", *name, err)
 		}
 
-		if tfJob.Status.State == torchv1alpha1.StateSucceeded || tfJob.Status.State == torchv1alpha1.StateFailed {
-			log.Infof("job %v finished:\n%v", *name, util.Pformat(tfJob))
+		if torchJob.Status.State == torchv1alpha1.StateSucceeded || torchJob.Status.State == torchv1alpha1.StateFailed {
+			log.Infof("job %v finished:\n%v", *name, util.Pformat(torchJob))
 			break
 		}
-		log.Infof("Waiting for job %v to finish:\n%v", *name, util.Pformat(tfJob))
+		log.Infof("Waiting for job %v to finish:\n%v", *name, util.Pformat(torchJob))
 		time.Sleep(5 * time.Second)
 	}
 
-	if tfJob == nil {
+	if torchJob == nil {
 		return *name, fmt.Errorf("Failed to get PyTorchJob %v", *name)
 	}
 
-	if tfJob.Status.State != torchv1alpha1.StateSucceeded {
+	if torchJob.Status.State != torchv1alpha1.StateSucceeded {
 		// TODO(jlewi): Should we clean up the job.
-		return *name, fmt.Errorf("PyTorchJob %v did not succeed;\n %v", *name, util.Pformat(tfJob))
+		return *name, fmt.Errorf("PyTorchJob %v did not succeed;\n %v", *name, util.Pformat(torchJob))
 	}
 
-	if tfJob.Spec.RuntimeId == "" {
+	if torchJob.Spec.RuntimeId == "" {
 		return *name, fmt.Errorf("PyTorchJob %v doesn't have a RuntimeId", *name)
 	}
 
+	// Loop over each replica and make sure the expected resources were created.
```

Comment:
You should be checking that the pods were created here right? You are looping over the replicaspec